weave_demo.ipynb

```diff
@@ -0,0 +1,1199 @@
+   "source": [
+    "# How to use Weave"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8cf03aa3-73db-4ece-91e4-ceeca931b81c",
+   "metadata": {},
+   "source": [
+    "Weave is a custom package used to facilitate the creation and maintenance of complex data warehouses.\n",
+    "\n",
+    "Weave was created from a need to track the lineage of data products derived from multiple sources. Weave can be used to upload arbitrary data products to a datastore with options to store metadata and information about how data were derived. When Weave is used to upload the data, Weave can then be used to access the data using Pythonic API calls, as well as giving the user easy access to data provenance.\n",
+    "\n",
+    "In this notebook we will demo some of weave's functionality."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "cba43194-5565-4be0-9ae5-0f4ea28b5e80",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "\n",
+    "import weave\n",
+    "from weave.pantry import Pantry\n",
+    "from weave.index.index_pandas import IndexPandas\n",
+    "from weave.index import index_abc\n",
+    "from weave import IndexSQLite\n",
+    "from weave import validate\n",
+    "\n",
+    "from fsspec.implementations.local import LocalFileSystem"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8afc1d77-2028-4a0f-aa1d-c82ea0589e45",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Creating a Pantry/Index\n",
+    "A pantry is a storage location that holds baskets or collections of baskets. An Index is an object or file that tracks the baskets in a pantry.\n",
+    "Below we will demonstrate how to create a pantry/Index, but first we need a location for the pantry. In this demo we will place all of our pantry's in the local file system. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "5f4b14a3-1ed7-4820-a744-a58543266eff",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "local_fs = LocalFileSystem()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "65f4187f-86b7-43c5-8104-7569d5cbabda",
+   "metadata": {},
+   "source": [
+    "Below we create a pantry using the local file system within our working directory. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "62e56228-39aa-4b07-9fb9-4aba8e055cde",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "pantry_name = \"weave-demo-pantry\"\n",
+    "local_fs.mkdir(pantry_name)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0fc0a17a-1046-4357-8c3d-34ca0a36e7bb",
+   "metadata": {},
+   "source": [
+    "Next let's create a pantry where we will store our basekts. For this pantry we will using Pandas for the index backend. An Index is an object or file that tracks the baskets in a pantry."
```

Comment:
Next, let's (comma needed). 

Also in the next sentence "For this pantry, we will..."

There's a missing "be" in the second sentence.

I think it makes sense to put the last sentence that describes an index as the second sentence. "Each pantry has an index this is an object... For this pantry, we..." This index line may not be needed because you use it just a few cells ago.